detection_rules/misc.py

```diff
@@ -32,6 +55,291 @@
+        self.__release_manifest_dict = dataclasses.asdict(self.release_manifest)
+        self.manifest_size = len(json.dumps(self.__release_manifest_dict))
+
+    @property
+    def release_manifest_fl(self):
+        return io.BytesIO(json.dumps(self.__release_manifest_dict, sort_keys=True).encode('utf-8'))
+
+    def _create(self):
+        """Create the manifest from GitHub asset metadata and file contents."""
+        assets = {}
+        for asset_name, asset_data in self.assets.items():
+            entries = {}
+            data = asset_data['data']
+            metadata = asset_data['metadata']
+
+            for file_name, file_data in data.items():
+                file_metadata = file_data['metadata']
+
+                name = Path(file_name).name
+                file_metadata.update(name=name)
+
+                entry = AssetManifestEntry(**file_metadata)
+                entries[name] = entry
+
+            assets[asset_name] = AssetManifestMetadata(metadata['browser_download_url'], entries,
+                                                       metadata['zipped_sha256'], metadata['created_at'],
+                                                       metadata['label'])
+
+        release_metadata = self._parse_release_metadata()
+        release_metadata.update(assets=assets)
+        release_manifest = ReleaseManifest(**release_metadata)
+
+        return release_manifest
+
+    def _parse_release_metadata(self):
+        """Parse relevant info from GitHub metadata for release manifest."""
+        ignore = ['assets']
+        manual_set_keys = ['author', 'description']
+        keys = [f for f in list(ReleaseManifest.__annotations__) if str(f) not in ignore + manual_set_keys]
+        parsed = {k: self.release.raw_data[k] for k in keys}
+        parsed.update(description=self.release.raw_data['body'], author=self.release.raw_data['author']['login'])
+        return parsed
+
+    def save(self) -> GitReleaseAsset:
+        """Save manifest files."""
+        if not self.has_token:
+            raise ValueError('You must provide a token to save a manifest to a GitHub release')
+
+        asset = self.release.upload_asset_from_memory(self.release_manifest_fl,
+                                                      self.manifest_size,
+                                                      self.manifest_name)
+        click.echo(f'Manifest saved as {self.manifest_name} to {self.release.html_url}')
+        return asset
+
+    @classmethod
+    def load(cls, name: str, repo: str = 'elastic/detection-rules', token=None) -> dict:
+        """Load a manifest."""
+        gh_client = GithubClient(token)
+        repo = gh_client.client.get_repo(repo)
+        release = get_gh_release(repo, tag_name=name)
+        asset = next((a for a in release.get_assets() if a.name == f'manifest-{name}.json'), None)
+        return load_json_gh_asset(asset.browser_download_url) if asset else None if asset else None
```

Comment:
this double ternary operator is very confusing and redundant.

can you just do this?
```suggestion
        if assert is not None:
            return load_json_gh_asset(asset.browser_download_url)
```

Reply:
lol I am not even sure what ..... nice catch

```suggestion
        if asset is not None:
            return load_json_gh_asset(asset.browser_download_url)
```